crates/starknet/src/starknet/add_deploy_account_transaction.rs

```diff
@@ -0,0 +1,175 @@
+
+        assert_eq!(txn.status, TransactionStatus::Rejected);
+    }
+
+    #[test]
+    fn deploy_account_transaction_successful_execution() {
+        let (mut starknet, addr, fee_token_address) = setup(None);
+        let account_class_hash = starknet.state.get_class_hash_at_contract_address(&addr).unwrap();
+
+        let transaction = DeployAccountTransaction::new(
+            vec![],
+            2000,
+            vec![],
+            Felt::from(0),
+            account_class_hash,
+            Felt::from(13),
+            DEVNET_DEFAULT_CHAIN_ID.to_felt().into(),
+            Felt::from(0),
+        )
+        .unwrap();
+
+        let account_address =
+            ContractAddress::try_from(transaction.0.contract_address().clone()).unwrap();
+        let balance_storage_var_address =
+            get_storage_var_address("ERC20_balances", &[account_address.into()]).unwrap();
+        let balance_storage_key =
+            ContractStorageKey::new(fee_token_address, balance_storage_var_address);
+
+        // change balance at address
+        let account_balance_before_deployment = Felt::from(1000000);
+        starknet
+            .state
+            .change_storage(balance_storage_key, account_balance_before_deployment)
+            .unwrap();
+        starknet.state.synchronize_states();
+
+        // get accounts count before deployment
+        let accounts_before_deployment = starknet.state.state.address_to_nonce.len();
+
+        let (txn_hash, _) = starknet.add_deploy_account_transaction(transaction).unwrap();
+        let txn = starknet.transactions.get_by_hash_mut(&txn_hash).unwrap();
+
+        assert_eq!(txn.status, TransactionStatus::AcceptedOnL2);
+        assert_eq!(starknet.state.state.address_to_nonce.len(), accounts_before_deployment + 1);
+        let account_balance_after_deployment =
+            starknet.state.get_storage(balance_storage_key).unwrap();
+
+        assert!(account_balance_before_deployment > account_balance_after_deployment);
+    }
+
+    /// Initializes starknet with 1 account - account without validations
+    fn setup(acc_balance: Option<u128>) -> (Starknet, ContractAddress, ContractAddress) {
+        let mut starknet = Starknet::default();
+        let account_json_path = concat!(
+            env!("CARGO_MANIFEST_DIR"),
+            "/test_artifacts/account_without_validations/account.json"
+        );
+        let contract_class = load_cairo_0_contract_class(account_json_path).unwrap();
+
+        let erc_20_contract = predeployed::create_erc20().unwrap();
+        erc_20_contract.deploy(&mut starknet.state).unwrap();
+
+        let acc = Account::new(
+            Felt::from(acc_balance.unwrap_or(100)),
+            dummy_felt(),
+            dummy_felt(),
+            contract_class.generate_hash().unwrap(),
+            contract_class,
+            erc_20_contract.get_address(),
+        )
+        .unwrap();
+
+        acc.deploy(&mut starknet.state).unwrap();
```

Comment:
Why are we deploying the account that we should later be deploying using the DeployAccount transaction?

Reply:
It was easier to deploy the account and then use the class hash of already deployed account to construct the deploy transaction

Reply:
But when thinking of it, its actually easier to just declare the class hash

Reply:
Changed in [95a0224](https://github.com/0xSpaceShard/starknet-devnet-rs/pull/73/commits/95a022461e1aed1f0f5f3434844cbc9eb52b1fd9)